javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/javassistmodel/JavassistAnnotationDeclaration.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2016 Federico Tomassetti
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.github.javaparser.symbolsolver.javassistmodel;
+
+import com.github.javaparser.resolution.declarations.*;
+import com.github.javaparser.resolution.types.ResolvedReferenceType;
+import com.github.javaparser.resolution.types.ResolvedType;
+import com.github.javaparser.symbolsolver.logic.AbstractTypeDeclaration;
+import com.github.javaparser.symbolsolver.model.resolution.TypeSolver;
+import javassist.CtClass;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * @author Malte Skoruppa
+ */
+public class JavassistAnnotationDeclaration extends AbstractTypeDeclaration implements ResolvedAnnotationDeclaration {
+
+    private CtClass ctClass;
+    private TypeSolver typeSolver;
+    private JavassistTypeDeclarationAdapter javassistTypeDeclarationAdapter;
+
+    @Override
+    public String toString() {
+        return getClass().getSimpleName() + "{" +
+               "ctClass=" + ctClass.getName() +
+               ", typeSolver=" + typeSolver +
+               '}';
+    }
+
+    public JavassistAnnotationDeclaration(CtClass ctClass, TypeSolver typeSolver) {
+        if (!ctClass.isAnnotation()) {
+            throw new IllegalArgumentException("Not an annotation: " + ctClass.getName());
+        }
+        this.ctClass = ctClass;
+        this.typeSolver = typeSolver;
+        this.javassistTypeDeclarationAdapter = new JavassistTypeDeclarationAdapter(ctClass, typeSolver);
+    }
+
+    @Override
+    public String getPackageName() {
+        return ctClass.getPackageName();
+    }
+
+    @Override
+    public String getClassName() {
+        String className = ctClass.getName().replace('$', '.');
+        if (getPackageName() != null) {
+            return className.substring(getPackageName().length() + 1, className.length());
+        }
+        return className;
+    }
+
+    @Override
+    public String getQualifiedName() {
+        return ctClass.getName().replace('$', '.');
+    }
+
+    @Override
+    public boolean isAssignableBy(ResolvedType type) {
+        throw new UnsupportedOperationException();
```

Comment:
Can we reuse the logic we have in the other Javassist classes?

Reply:
This is a problem that already exists in `JavaParserAnnotationDeclaration`. There too, this method simply throws an `UnsupportedOperationException`. I would suggest that, as @matozoid asked, I make an issue for each comment here where `JavaParserAnnotationDeclaration` also simply throws an `UnsupportedOperationException`, because that means that it is not a new issue, but something that is already a missing functionality now. That way, we can think about what we want to do in each case, and then this logic must be implemented for all *three* classes `JavaParserAnnotationDeclaration`, `ReflectionAnnotationDeclaration`, and `JavassistAnnotationDeclaration`, not only the two new ones which I added. Is that ok?

So... I'll make an issue about this one.

Reply:
See issue #1836.